synapse/rest/media/v1/media_storage.py

```diff
@@ -104,7 +105,11 @@ def store_into_file(self, file_info):
 
         async def finish():
             for provider in self.storage_providers:
-                await provider.store_file(path, file_info)
+                # store_file is supposed to return an Awaitable or Deferred, but
+                # guard against improper implementations.
+                result = provider.store_file(path, file_info)
+                if inspect.isawaitable(result) or isinstance(result, defer.Deferred):
```

Comment:
I think Deferred is awaitable.

Reply:
Oh, right. That's why using `await` with it works! 🤦